web/src/main/java/org/fao/geonet/kernel/mef/MEFVisitor.java

```diff
@@ -65,13 +77,23 @@ public Element handleXml(File mefFile, IMEFVisitor v) throws Exception {
 
 		try {
 			while ((entry = zis.getNextEntry()) != null) {
-				String name = entry.getName();
 
-				if (name.equals(FILE_METADATA))
-					md = Xml.loadStream(isb);
-				else if (name.equals(FILE_INFO))
-					info = Xml.loadStream(isb);
+				if(entry.isDirectory()){
+					continue;
+				}
 
+				String fullName = entry.getName();
+				String simpleName = new File(fullName).getName();
+
+				if(this.recordInfo.schema.startsWith(this.preferredSchema)) {
```

Comment:
Good catch on the null pointer exception, I forgot about that one! 

The `metadataFile` string is a good idea however I still require the `preferredSchema` in the function as well as the `recordInfo` object as I actually modify `recordInfo.schema` to prevent unknown schema issues in other parts of the code. Since the metadata is in fact now a 19139 record the original `recordInfo` is now not a valid representation, hence the overwrite.